### PR TITLE
xmr/bp: memory optimizations and improvements, fixed style

### PR DESCRIPTION
**Pull request by onvej-sl**
This PR supersedes https://github.com/trezor/trezor-firmware/pull/717.

### CODE REVIEW

File: core/tests/test_apps.monero.bulletproof.py [hunk 1]
```diff
@@ -285,102 +285,27 @@ def test_masks(self):
         bpi.use_det_masks = False
         self.mask_consistency_check(bpi)
 
-    def test_verify_testnet(self):
```
Comment (prusnak) [inline core/tests/test_apps.monero.bulletproof.py]: Why is this test removed?

File: core/tests/test_apps.monero.bulletproof.py [hunk 2]
```diff
@@ -403,15 +328,6 @@ def test_prove_random_masks(self):
         bp_res = bpi.prove(val, mask)
         bpi.verify(bp_res)
 
-    def test_prove_testnet_random_masks(self):
```
Comment (prusnak) [inline core/tests/test_apps.monero.bulletproof.py]: Why is this test removed?

File: core/tests/test_apps.monero.bulletproof.py (hunk 1 above)
Comment (onvej-sl) [reply core/tests/test_apps.monero.bulletproof.py]: This type of proofs is no longer supported. See `verify_testnet` in `bulletproof.py ` and the commit message: 
> proof_v8 support discontinued, old hardfork, not needed anymore

I think it has something to do with [the old vulnerability](https://www.getmonero.org/2017/05/17/disclosure-of-a-major-bug-in-cryptonote-based-currencies.html).

File: core/tests/test_apps.monero.bulletproof.py (hunk 2 above)
Comment (onvej-sl) [reply core/tests/test_apps.monero.bulletproof.py]: See above.

**Pull request merged**